src/java.base/share/classes/jdk/internal/classfile/components/ClassRemapper.java

```diff
@@ -0,0 +1,401 @@
+                    case InvokeInstruction ii ->
+                        cob.invokeInstruction(ii.opcode(), map(ii.owner().asSymbol()), ii.name().stringValue(), mapMethodDesc(ii.typeSymbol()), ii.isInterface());
+                    case InvokeDynamicInstruction idi ->
+                        cob.invokeDynamicInstruction(DynamicCallSiteDesc.of(idi.bootstrapMethod(), idi.name().stringValue(), mapMethodDesc(idi.typeSymbol())));
+                    case NewObjectInstruction c ->
+                        cob.newObjectInstruction(map(c.className().asSymbol()));
+                    case NewReferenceArrayInstruction c ->
+                        cob.anewarray(map(c.componentType().asSymbol()));
+                    case NewMultiArrayInstruction c ->
+                        cob.multianewarray(map(c.arrayType().asSymbol()), c.dimensions());
+                    case TypeCheckInstruction c ->
+                        cob.typeCheckInstruction(c.opcode(), map(c.type().asSymbol()));
+                    case ExceptionCatch c ->
+                        cob.exceptionCatch(c.tryStart(), c.tryEnd(), c.handler(), c.catchType().map(d -> TemporaryConstantPool.INSTANCE.classEntry(map(d.asSymbol()))));
+                    case LocalVariable c ->
+                        cob.localVariable(c.slot(), c.name().stringValue(), map(c.typeSymbol()), c.startScope(), c.endScope());
+                    case LocalVariableType c ->
+                        cob.localVariableType(c.slot(), c.name().stringValue(), mapSignature(c.signatureSymbol()), c.startScope(), c.endScope());
+                    case LoadConstantInstruction ldc ->
+                        cob.constantInstruction(ldc.opcode(), mapConstantValue(ldc.constantValue()));
+                    case RuntimeVisibleTypeAnnotationsAttribute aa ->
+                        cob.with(RuntimeVisibleTypeAnnotationsAttribute.of(mapTypeAnnotations(aa.annotations())));
+                    case RuntimeInvisibleTypeAnnotationsAttribute aa ->
+                        cob.with(RuntimeInvisibleTypeAnnotationsAttribute.of(mapTypeAnnotations(aa.annotations())));
+                    default ->
+                        cob.with(coe);
+                }
+            };
+        }
+
+        @Override
+        public ClassDesc map(ClassDesc desc) {
+            if (desc == null) return null;
+            if (desc.isArray()) return map(desc.componentType()).arrayType();
+            if (desc.isPrimitive()) return desc;
+            return mapFunction.apply(desc);
+        }
+
+        MethodTypeDesc mapMethodDesc(MethodTypeDesc desc) {
+            return MethodTypeDesc.of(map(desc.returnType()), desc.parameterList().stream().map(this::map).toArray(ClassDesc[]::new));
+        }
+
+        ClassSignature mapClassSignature(ClassSignature signature) {
+            return ClassSignature.of(signature.typeParameters(),
```

Comment:
Should type parameters also be mapped? (as they might have class bounds). Both here and in `mapMethodSignature`.

Reply:
Nice catch, will fix it, thanks!